src/main/java/gov/usgs/wma/waterdata/LoadDiscreteGroundWater.java

```diff
@@ -33,39 +33,44 @@ public  ResultObject apply(RequestObject request) {
 
 	@Transactional
 	protected ResultObject processRequest(RequestObject request) {
-		List<String> fieldVisitIdentifiers = request.getFieldVisitIdentifiers();
-		LOG.debug("the request object: {}", fieldVisitIdentifiers.toString());
+
+		String locationIdentifier = request.getLocationIdentifier();
+		LOG.debug("the request object location id: {}", locationIdentifier);
+
 		ResultObject result = new ResultObject();
 
-		for (String fieldVisitIdentifier : fieldVisitIdentifiers) {
-			// each field visit identifier could have more than one record returned
-			List<DiscreteGroundWater> discreteGroundWaterList = transformDao.getDiscreteGroundWater(fieldVisitIdentifier);
-			int rowsInserted = loadDiscreteGroundWaterIntoObservationDb(discreteGroundWaterList, fieldVisitIdentifier);
-			if (null == result.getCount()) {
-				result.setCount(rowsInserted);
-			} else {
-				result.setCount(result.getCount() + rowsInserted);
-			}
+		// get all the records for this location identifier from the transform db
+		List<DiscreteGroundWater> discreteGroundWaterList = transformDao.getDiscreteGroundWater(locationIdentifier);
+
+		int rowsInserted = 0;
+
+		// if we do not get records back from the transform db, don't try to load anything to the observation db
+		if (discreteGroundWaterList.size() > 0) {
+			rowsInserted = loadDiscreteGroundWaterIntoObservationDb(discreteGroundWaterList);
 		}
+
+		result.setCount(rowsInserted);
```

Comment:
We may want to expand the Result concept.  It just just inserted rows, there are also deleted and updated records.